packages/features/schedules/components/DateOverrideInputDialog.tsx

```diff
@@ -0,0 +1,189 @@
+import { useState, useEffect, useMemo } from "react";
+import { useForm } from "react-hook-form";
+
+import dayjs, { Dayjs } from "@calcom/dayjs";
+import { daysInMonth, yyyymmdd } from "@calcom/lib/date-fns";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { WorkingHours } from "@calcom/types/schedule";
+import {
+  Dialog,
+  DialogContent,
+  DialogTrigger,
+  DialogHeader,
+  DialogClose,
+  Switch,
+  DatePicker,
+  Form,
+  Button,
+} from "@calcom/ui";
+
+import { DayRanges, TimeRange } from "./Schedule";
+
+const ALL_DAY_RANGE = {
+  start: new Date(dayjs.utc().hour(0).minute(0).second(0).format()),
+  end: new Date(dayjs.utc().hour(0).minute(0).second(0).format()),
+};
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+
+const DateOverrideForm = ({
+  value,
+  workingHours,
+  excludedDates,
```

Comment:
What if there's a use case where someone's availability is Monday - Thursday but they want to open up some dates on a Friday? Right now we can't choose dates that are not a part of the original schedule.

Reply:
Yes, the scope of this feature is to override available dates - it's not intended to open up dates. That's a future improvement - I tried to embed that in the current design but honestly I shouldn't have and it didn't work right.

Reply:
while it would be great if we could do what you're stating @joeauyeung, fundamentally we're overriding availability ... not overriding unavailability. If people want to make some Fridays available, we do have the ability to create multiple availability schedules, you can make a new one.